pkg/virt-controller/watch/vm.go

```diff
@@ -1111,6 +1120,15 @@ func isSetToStart(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualMachineInstance)
 	}
 }
 
+func (c *VMController) cleanupRestartRequired(vm *virtv1.VirtualMachine) error {
+	vmConditionManager := controller.NewVirtualMachineConditionManager()
+	if vmConditionManager.HasCondition(vm, virtv1.VirtualMachineRestartRequired) {
+		vmConditionManager.RemoveCondition(vm, virtv1.VirtualMachineRestartRequired)
+	}
+
+	return c.deleteVMRevisions(vm, revisionPrefixLastSeen)
```

Comment:
the `deleteVMRevisions` can be dropped.

Reply:
https://github.com/kubevirt/kubevirt/issues/11151